src/lib.rs

```diff
@@ -1,25 +1,73 @@
 use itertools::*;
 use lazy_static::*;
+use rand::distributions::*;
+use rand::{CryptoRng, Rng};
 use regex::Regex;
 use sha2::{Digest, Sha256};
 use std::collections::HashSet;
+use std::ops::DerefMut;
+use std::sync::{Mutex, MutexGuard};
 use unidecode::unidecode_char;
 use voca_rs::*;
+use Result::{Err, Ok};
 
 lazy_static! {
     ///Special chars that should be filtered out.
     static ref SPECIAL_CHAR: Regex = Regex::new(r#"[!@#$%^&*(){}_<>:;,."'`|+=/~\[\]\\-]"#).unwrap_or_else(|e| panic!("Developer error. Bad regex {:?}", e));
+    static ref ALL_U32: Uniform<u32> = Uniform::new_inclusive(0u32, u32::max_value());
```

Comment:
Having this as a uniform distribution makes the `sample_iter` below much nicer. Otherwise I'd have to implement that with a loop which I find less clear.